httpserver/http_connection.cpp

```diff
@@ -140,12 +148,15 @@ private_key_t LokidClient::wait_for_privkey() {
                 throw std::runtime_error("empty body");
             else {
                 auto r = nlohmann::json::parse(*res.body);
-                const auto &privkey = r.at("result").at("service_node_privkey").get_ref<const std::string &>();
-                if (privkey.size() != 2 * loki::KEY_LENGTH && !std::all_of(privkey.begin(), privkey.end(),
-                            [](char c) { return (c >= '0' && c <= '9') || (c >= 'a' && c <= 'f'); }))
+                const auto &legacy_privkey = r.at("result").at("service_node_privkey").get_ref<const std::string &>();
+                const auto &privkey_ed = r.at("result").at("service_node_ed25519_privkey").get_ref<const std::string &>();
+                const auto &privkey_x = r.at("result").at("service_node_x25519_privkey").get_ref<const std::string &>();
+                if (!validateHexKey(legacy_privkey) || !validateHexKey(privkey_ed) || !validateHexKey(privkey_x))
                     throw std::runtime_error("returned value is not hex");
                 else {
-                    private_key = loki::lokidKeyFromHex(privkey);
+                    private_key = loki::lokidKeyFromHex(legacy_privkey);
+                    private_key_ed = private_key_ed25519_t::from_hex(privkey_ed);
+                    private_key_x = loki::lokidKeyFromHex(privkey_x);
```

Comment:
Possibly want to sanity check that the x25519 key is derived from the given edkeys? Not sure what role the x25519 keys play in storage server.

Reply:
Added as a todo.